src/arithmetic/numeric_funcs/numeric_funcs.c

```diff
@@ -374,7 +374,7 @@ void Register_NumericFuncs() {
 	AR_FuncDesc *func_desc;
 
 	types = array_new(SIType, 1);
-	array_append(types, (SI_NUMERIC | T_STRING | T_ARRAY | T_BOOL | T_NULL));
+	array_append(types, (SI_NUMERIC | T_STRING | T_ARRAY | T_BOOL | T_NODE | T_NULL));
```

Comment:
why not use SI_ALL?

Reply:
what happen if adding int to node?

Reply:
I am not sure if all the types listed in `SI_ALL` Neo4j supports to be added to a list. I'll check.

Reply:
> what happen if adding int to node?

Overlooked, fixed. Thanks for noticing this.

Reply:
As said, still checking. I have already confirmed that edge works in Neo4J. I'll update you once I finish checking.

Reply:
So paths also work. Now I really wonder whether we can use all objects here or not cuz I guess we should be able to.

Reply:
Changed to SI_ALL. Neo4J seems to be easily working with any type of object when adding to a list. However, this same function is used for addition with other types, so I had to blacklist paths, nodes and edges from being added to anything else, which also is logical and conforms with Neo4J behaviour. In the meantime though, I have found that:

- `true + 2` causes an error in Neo4J, but in RG works (result 3).
- `point() + 2` causes an error in Neo4J, but in RG works (the result is some number).

Other types like datetime/time/duration/etc either aren't supported by RG (though defined in the enum) and/or work as in Neo4J.

Reply:
Forbade addition of a point with any other type rather than an array as well. Leaving boolean additions in place as it would be a breaking behaviour (discussed privately).